docs/source/debugging.md

```diff
@@ -0,0 +1,183 @@
+and can be used as extra command line arguments with the [hi-ml
+runner](https://github.com/microsoft/hi-ml/blob/746c8b58c1af71f71eeaaac2a8584be1d9a5386f/hi-ml/src/health_ml/runner.py#L107).
+
+* `pl-fast-dev-run`: If set to `n`, runs the pipeline for only `n` batch(es) of train, val and test for only a single
+  epoch. Additionally [this flag](https://pytorch-lightning.readthedocs.io/en/stable/common/debugging.html#fast-dev-run)
+  disables all callbacks and hyperparameters serialization which makes the debugging process very quick. This must be
+  used for debugging purposes only.
+* `pl-limit-train-batches`: Limits the training dataset to the given number of batches `n`.
+* `pl-limit-val-batches`: Limits the validation dataset to the given number of batches `n`.
+* `pl-limit-train-batches`: Limits the test dataset to the given number of batches `n`.
+
+In general, it is very useful to run the following two steps as part of the developement cycle:
+
+1. Make sure all training, validation and test loops complete properly:
+
+```shell
+conda activate HimlHisto
+python ../hi-ml/src/health_ml/runner.py --model=YourCustomContainer --crossval-count=0 --bach-size=2 --pl-fast-dev-run=4
+```
+
+2. Make sure the whole pipeline runs properly, including checkpoints callbacks and hyperparameter serialization:
+
+```shell
+conda activate HimlHisto
+python ../hi-ml/src/health_ml/runner.py --model=YourCustomContainer --crossval-count=0 --bach-size=2 --pl-limit-train-batches=4 --pl-limit-val-batches=4 --pl-limit-test-batches=4 --max_epochs=4
+```
+
+Note: Under the hood, setting `pl-fast-dev-run=n` overrides
+`pl-limit-train-batches=n`, `pl-limit-val-batches=n`, `pl-limit-train-batches=n`, `max_epochs=1` and disables all
+callbacks. Please keep in mind that all the above is useful for efficient and quick debugging purposes only.
+
+## Profiling Machine Learning Pipelines
+
+Pytorch Lightning supports a set of built-in
+[profilers](https://pytorch-lightning.readthedocs.io/en/stable/advanced/profiler.html) that help you identify
+bottlenecks in your code during training, testing and inference. You can trigger code profiling through the command line
+argument `--pl_profiler` that you can set to either
+[`simple`](https://pytorch-lightning.readthedocs.io/en/stable/advanced/profiler.html#simple-profiler),
+[`advanced`](https://pytorch-lightning.readthedocs.io/en/stable/advanced/profiler.html#simple-profiler), or
+[`pytorch`](https://pytorch-lightning.readthedocs.io/en/stable/advanced/profiler.html#pytorch-profiler).
+
+The profiler outputs will be saved in a subfolder `profiler` inside the outputs folder of the run.
+
+```shell
+conda activate HimlHisto
+python ../hi-ml/src/health_ml/runner.py --model=YourCustomContainer --crossval-count=0 --bach-size=2 --pl-limit-train-batches=4 --pl-limit-val-batches=4 --pl-limit-test-batches=4 --max_epochs=4 --pl-profiler=pytorch
```

Comment:
It's not immediate obvious to the reader which is the important flag here. Perhaps we could instead say 'add the following flag to your command' and just show the flag of interest

Reply:
I just wanted to make it clear that these flags should be used along with the himl runner

Reply:
You could explain that with a sentence tho rather than repeating the runner command